core/ui/vault/chain/manage/index.tsx

```
@@ -0,0 +1,132 @@
+import { chainFeeCoin } from '@core/chain/coin/chainFeeCoin'
+import { areEqualCoins, Coin } from '@core/chain/coin/Coin'
+import { ChainEntityIcon } from '@core/ui/chain/coin/icon/ChainEntityIcon'
+import { getChainEntityIconSrc } from '@core/ui/chain/coin/icon/utils/getChainEntityIconSrc'
+import {
+  useCreateCoinMutation,
+  useDeleteCoinMutation,
+} from '@core/ui/storage/coins'
+import { useCurrentVaultNativeCoins } from '@core/ui/vault/state/currentVaultCoins'
+import { Switch } from '@lib/ui/inputs/switch'
+import { TextInput } from '@lib/ui/inputs/TextInput'
+import { HStack, VStack } from '@lib/ui/layout/Stack'
+import { List } from '@lib/ui/list'
+import { ListItem } from '@lib/ui/list/item'
+import { ListItemTag } from '@lib/ui/list/item/tag'
+import { PageContent } from '@lib/ui/page/PageContent'
+import { PageHeader } from '@lib/ui/page/PageHeader'
+import { PageHeaderBackButton } from '@lib/ui/page/PageHeaderBackButton'
+import { PageHeaderTitle } from '@lib/ui/page/PageHeaderTitle'
+import { Text } from '@lib/ui/text'
+import { FC, useMemo, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+
+const NativeCoinItem: FC<Coin> = coin => {
+  const currentCoins = useCurrentVaultNativeCoins()
+  const createCoin = useCreateCoinMutation()
+  const deleteCoin = useDeleteCoinMutation()
+
+  const currentCoin = useMemo(() => {
+    return currentCoins.find(c => areEqualCoins(c, coin))
+  }, [currentCoins, coin])
+
+  const handleChange = async () => {
+    if (currentCoin) {
+      await deleteCoin.mutateAsync(currentCoin)
```

I think better to use regular mutate here, and in general as well unless there is another action depending on it, because mutateAsync should throw an error, while the regular mutate will just registed the error in the hook result, plus here we don't actually do anything after the await